week_1/level2_assignment/random.ts

```diff
@@ -0,0 +1,57 @@
+// 과제 조건
+// 1. Member, Dinner interface 만들고 타입 지정하기
+// 2. organize 내부 로직 채우기
+
+interface Member {
+    name: string;
+    group: string;
+}
+
+interface Dinner {
+    member: Array<Member>;
+    shuffle(array_for_shu: Array<Member>): Array<Member>;
+    organize(array_for_org: Array<Member>): void;
+}
+
+const dinner: Dinner = {
+    member: [
+        {
+            name: '채정아',
+            group: 'ob'
+        },
+        {
+            name: '김동재',
+            group: 'yb'
+        },
+        {
+            name: '강민재',
+            group: 'yb'
+        },
+        {
+            name: '김루희',
+            group: 'ob'
+        },
+        {
+            name: '박진수',
+            group: 'ob'
+        }
+    ],
+
+    shuffle(array) {
+        array.sort(() => Math.random() - 0.5);
+        return array;
+    },
+
+    organize(array) {
+        this.shuffle(array);
+        let dinnerMember: Array<string> = [];
+        let temp: Member = array.find((element) => (element.group === 'ob'))!; // 느낌표 없으면 ts2322 오류 발생 
+        dinnerMember[0] = temp.name;
+        temp = array.find((element) => (element.group === 'yb'))!;
+        dinnerMember[1] = temp.name;
```

Comment:
이것도 좋지만 무엇보다도 코드를 줄이는게 개발자로선 최고의 선택이니..
저는 요렇게 해결하긴 했습니다만 저도 알고리즘을 잘하진 못해서 .. ㅋㅋ ㅋ 참고만 해주시면 감사하겠심다~

그리고 사용하신 느낌표는 세미나 때 배운 as 타입 단언을 하는것입니다!
해당 오류를 해결하는데는 좋지만 남발하지 않는 것을 권장한다고 하네요... (여기)[https://kyounghwan01.github.io/blog/TS/object-null/]를 참고해보시면 좋을듯..! 여기서는 타입단언이나 if문, 옵셔널체이닝을 해결방법으로 소개하는데 저도 옵셔널체이닝 방법으로 해결했슴니다.!

```suggestion
    const ob = array.filter((o) => o.group === "ob").pop();
    const yb = array.filter((o) => o.group === "yb").pop();

    console.log(`오늘의 저녁 식사 멤버는 ${ob?.name}, ${yb?.name}`);
```

Reply:
저부분 코드가 더러워서 찝찝했는데🥲 세훈님 말씀대로 바꿔보겠습니다 ! 덜 사용하는 것도 기억할게요 꼼꼼한 리뷰 땡큐베리머치🥺🥺